analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.IsNull.cs

```diff
@@ -205,5 +205,27 @@ public void IsNull_ConditionalAccess_OnUnknown()
                 "WasNotNull",
                 "End");
         }
+
+        [TestMethod]
+        public void IsNull_TryCast_UpCast()
+        {
+            var validator = SETestContext.CreateCS("""
+                (arg as Exception)?.ToString();
+                Tag("Arg", arg);
+                """, ", object arg").Validator;
+            validator.ValidateTag("Arg", x => x.Should().HaveNoConstraints());
+        }
+
+        [TestMethod]
+        public void IsNull_TryCast_DownCast()
+        {
+            var validator = SETestContext.CreateCS("""
+                (arg as Exception)?.ToString();
+                Tag("Arg", arg);
+                """, ", ArgumentException arg").Validator;
+            validator.TagValues("Arg").Should().SatisfyRespectively(
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.Null),
+                x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+        }
```

Comment:
The terminology is mixed. The first test is a downcast, and the second is an upcast.